packages/orbit-components/src/primitives/ButtonPrimitive/index.tsx

```diff
@@ -1,263 +1,297 @@
+      // Content vars
       contentWidth,
+      // Icon vars
+      icons,
+      // Modificators
       fullWidth,
       centered,
-    } = props;
-    const { width, height, leftMargin, rightMargin } = icons;
-    const isDisabled = loading || disabled;
-    const onlyIcon = Boolean(iconLeft && !children);
+      underlined,
+      circled,
+      spaceAfter,
+      // Status
+      loading,
+      disabled,
+      // Rest
+      className,
+      ...props
+    }: Props,
+    ref,
+  ) => {
+    const isLink = props.href != null || props.asComponent === "a";
+    const isDisabled = loading === true || disabled === true;
+
+    const varsButton = {
+      "--button-height": height,
+      "--button-padding": padding,
+      "--button-content-align": contentAlign,
+      "--button-font-weight": fontWeight,
+      "--button-font-size": fontSize,
+      "--button-foreground": foreground,
+      "--button-foreground-hover": foregroundHover,
+      "--button-foreground-active": foregroundActive,
+      "--button-foreground-focus": foregroundFocus,
+      "--button-background": background,
+      "--button-background-hover": backgroundHover,
+      "--button-background-active": backgroundActive,
+      "--button-background-focus": backgroundFocus,
+      "--button-box-shadow": boxShadow,
+      "--button-box-shadow-hover": boxShadowHover,
+      "--button-box-shadow-focus": boxShadowFocus,
+      "--button-box-shadow-active": boxShadowActive,
+      // loader
+      "--button-icon-width": icons?.width,
+      "--button-icon-height": icons?.height,
+    } as React.CSSProperties;
+
+    const varsContent = {
+      "--button-content-width": contentWidth,
+    } as React.CSSProperties;
+
+    const varsIcons = {
+      "--button-icon-width": icons?.width,
+      "--button-icon-height": icons?.height,
+      "--button-icon-left-margin": icons?.leftMargin,
+      "--button-icon-right-margin": icons?.rightMargin,
+      "--button-icon-foreground": icons?.foreground,
+      "--button-icon-foreground-focus": icons?.foregroundFocus,
+      "--button-icon-foreground-hover": icons?.foregroundHover,
+      "--button-icon-foreground-active": icons?.foregroundActive,
+    } as React.CSSProperties;
 
     return (
-      <StyledButtonPrimitive ref={ref} onlyIcon={onlyIcon} {...props} disabled={isDisabled}>
+      <ButtonPrimitiveComponent
+        ref={ref}
+        {...props}
+        disabled={isDisabled}
+        className={cx(
+          className,
+          "orbit-button-primitive font-base duration-fast group relative max-w-full cursor-pointer items-center border-none text-center transition-all [&>*]:align-middle [&_.orbit-loading-spinner]:stroke-[currentColor]",
+          fullWidth && "w-full",
+          centered || children == null ? "justify-center" : "justify-[var(--button-content-align)]",
+          circled !== true && "rounded-large tb:rounded-normal",
+          isDisabled
+            ? "opacity-button-disabled cursor-not-allowed"
```

Comment:
Non-blocking but just wondering if we really need this token... We have `opacity-30` available, and it's much more explicit and one less token for us...

Reply:
imo we don't, but again, it exists so I used it. gonna remove its usage in TW and Button, we can clean it up later

Reply:
It's part of the component tokens, so it can be removed with no breaking change notice. I'd just remove it now and leave only the deprecated one, to be removed on the next major tokens release. But up to you 👍